docs/root/start/sandboxes/brotli.rst

```diff
@@ -0,0 +1,102 @@
+.. _install_sandboxes_brotli:
+
+Brotli
+======
+
+.. sidebar:: Requirements
+
+   .. include:: _include/docker-env-setup-link.rst
+
+   :ref:`curl <start_sandboxes_setup_curl>`
+        Used to make ``HTTP`` requests.
+
+By enabling compression in Envoy you can save some network bandwidth, at the expense of increased processor usage.
+
+Envoy supports compression and decompression for both requests and responses.
+
+This sandbox provides an example of response compression served over ``HTTP``. Although ``HTTPS`` is not demonstrated, compression can be used for this also.
+
+The sandbox covers two scenarios:
+
+- compression of files from an upstream server
+- compression of Envoy's own statistics
+
+Step 1: Start all of our containers
+***********************************
+
+Change to the ``examples/brotli`` directory and bring up the docker composition.
+
+.. code-block:: console
+
+    $ pwd
+    envoy/examples/brotli
+    $ docker-compose build --pull
+    $ docker-compose up -d
+    $ docker-compose ps
+    Name                 Command                        State   Ports
+    --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+    brotli_envoy-stats_1   /docker-entrypoint.sh /usr ... Up      0.0.0.0:10000->10000/tcp,:::10000->10000/tcp, 0.0.0.0:9901->9901/tcp,:::9901->9901/tcp, 0.0.0.0:9902->9902/tcp,:::9902->9902/tcp
+    brotli_service_1       python3 /code/service.py       Up
+
+Step 2: Test Envoy’s compression of upstream files
+**************************************************
+
+The sandbox is configured with two endpoints on port ``10000`` for serving upstream files:
+
+- ``/file.txt``
+- ``/file.json``
+
+Only ``/file.json`` is configured to be compressed.
+
+Use ``curl`` to check that the response from requesting ``file.json`` contains the ``content-encoding: br`` header.
+
+You will need to add an ``accept-encoding: br`` request header.
+
+.. code-block:: console
+
+    $ curl -si -H "Accept-Encoding: br" localhost:10000/file.json | grep "content-encoding"
+    content-encoding: br
+
+As only files with a content-type of ``application/json`` are configured to be compressed, the response from requesting ``file.txt`` should not contain the ``content-encoding: br`` header, and the file will not be compressed:
+
+.. code-block:: console
+
+    $ curl -si -H "Accept-Encoding: br" localhost:10000/file.txt | grep "content-encoding"
+
+It still works when specify mulitiple compressor filters:
+
+.. code-block:: console
+
+    $ curl -si -H "Accept-Encoding: gzip;q=0.5, br" localhost:10000/file.json | grep "content-encoding"
+    content-encoding: br
+
```

Comment:
Hmm... in hindsight this doesn't look like a good idea after all. The example config contains only one compression filter per filter chain and mentioning gzip in this context may confuse readers. Better drop these lines.

Requests with `Accept-Encoding: gzip;q=0.5, br` could be illustrative for setups featuring more than one compression filter in a filter chain.

Reply:
these examples should be as simple as possible and provide an isolated setup that is exemplary of what is being shown

i think in this case we just want the `br` `accept-encoding` header

Reply:
Remove them.

Reply:
sorry for force push :(